documentation/contribute/contributingDatabaseSupport.md

```diff
@@ -15,54 +15,76 @@ We welcome pull requests for database support, but not all requests are automati
-*   `flyway.locations` - to point to an accessible folder where you can put test migrations.
+repositories {
+    mavenCentral()
+}
 
-You can now set up a run configuration in your IDE that will compile Flyway and run using your newly created configuration:
+dependencies {
+    compile "org.flywaydb:flyway-core:${flywayVersion}"
+    compile "${jdbcDriverDependency}"
+}
 
-*   Main class: `org.flywaydb.commandline.Main`
-*   Program arguments: `info -X -configFiles=<scratch location>\flyway.conf`
-*   Classpath of module: `flyway-commandline`
+flyway {
+    url = "${databaseUrl}"
+    user = "${databaseUser}"
+    password = "${databasePassword}"
+    locations = ["classpath:db/migration"]
+}
 
-Flyway itself should start. Since Flyway doesn’t yet support your database you should see a message like:
+flywayInfo.dependsOn classes
+flywayClean.dependsOn classes
+flywayMigrate.dependsOn classes
+```
 
-`org.flywaydb.core.api.FlywayException: Unable to autodetect JDBC driver for url: jdbc:mydatabase://<host>:<port>/<databasename>`
+You will need to fill in the `${flywayVersion}` with the latest version of flyway, `${jdbcDriverDependency}` with the jdbc driver dependency you wish to use, and everything in the `flyway` block with the url and credentials of your database.
 
-You’re now ready to start adding that database support. We’re going to assume your database platform is called **FooDb**. Change the obvious naming conventions to suit your database.
+At this point you can run `flywayMigrate` (either via `./gradlew flywayMigrate` or via the IDE), and gradle should fail with a message like `No database found to handle ${databaseUrl}`.
 
-## Let's code!
+Now we are ready to begin developing the actual database support.
 
-Here are all the changes and additions you'll need to make:
+### Developing support
 
-1.  Add an optional dependency to your JDBC driver to the top-level `.pom`
-1.  Add a dependency of your JDBC driver to the flyway-commandline `.pom` if you wish the driver to be automatically shipped with the Flyway product.
-1.  In `flyway.conf`, document the format of the JDBC connection url for your database. This is not necessary to make Flyway work but it will help adoption of your database!
-1.  Create a new folder `Foo` in `org.flywaydb.core.internal.database` to contain your new classes.
+We’re going to assume your database platform is called **FooDb**. Change the obvious naming conventions to suit your database.
+1. Flyway automatically scans the `db/database` package for classes that inherit from `DatabaseType`. Therefore in the `src/main/java` folder create a new package called something like `db/database/foodb`.
 1.  In the folder create classes `FooDatabase` (subclassed from Database), `FooSchema` (subclassed from Schema), and `FooTable` (subclassed from Table), using the canonical signatures. These classes make up Flyway’s internal representation of the parts of your database that it works on.
```

Comment:
Might want this to say the `foodb` folder or `this new` folder as _the folder_ refers to `src/main/java` in the previous sentence. Could stop someone from getting to the end and realizing they did it wrong:)